app/crud/metric_info.py

```diff
@@ -0,0 +1,36 @@
+import sqlalchemy
+from sqlalchemy.orm import Session
+from ..schemas import serviceMetric , metric_info , scoreCardMetrics,MetricInfoBase
+from .base import CRUDBase
+from . import CRUDServiceMetric , CRUDMetric ,CRUDScoreCardMetric
+
+
+
+class CRUDMetricInfo:
+    def __init__(self, db_session: Session):
+              self.servicemetric = CRUDServiceMetric(db_session)
+              self.Metric = CRUDMetric(db_session)
+              self.scorecardmetric = CRUDScoreCardMetric(db_session)
+              
+     
+    def get_latest_metric_readings(self, service_id: int, scorecard_id: int) -> list[MetricInfoBase]:
+      #metricweights= self.scorecardmetric.getMetricWeight(scorecard_id)
+      metricweights = {weight.metricId: weight.weight for weight in self.scorecardmetric.getMetricWeight(scorecard_id)}
```

Comment:
instead of using `getMetricWeight` for each metric record.
use the `get_metric` which will query the DB only once and return list of metric objects that contains the weight information, you can later reduce this list of objects to extract the weight information. 
It's better to query the DB once than querying it multiple times.